lib/resultuploaders/result_uploader.rb

```diff
@@ -10,24 +11,48 @@ class ResultUploader
     # UploaderFactory
     #
     class UploaderFactory
-      UPLOADERS = {
-        dropbox: Dropbox
-      }.freeze
+      def self.initialize(logger)
+        @uploades = {}
+
+        Dir['./lib/resultuploaders/*.rb'].sort.each do |file|
+          uploader_name = File.basename(file, '.rb')
+          # Ignore current file
+          next if uploader_name == File.basename(__FILE__, '.rb')
```

Comment:
What about creating a new directory `driver` and putting each uploader implementations into that directory? This will simplify the logic and clarifies this file and the other "driver" files have different roles.